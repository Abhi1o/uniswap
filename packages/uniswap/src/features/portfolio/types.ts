import { CurrencyId } from 'uniswap/src/types/currency'

export const HIDDEN_TOKEN_BALANCES_ROW = 'HIDDEN_TOKEN_BALANCES_ROW' as const

export function isHiddenTokenBalancesRow(row: string): row is typeof HIDDEN_TOKEN_BALANCES_ROW {
  return row === HIDDEN_TOKEN_BALANCES_ROW
}

export type TokenBalanceListRow = CurrencyId | typeof HIDDEN_TOKEN_BALANCES_ROW
