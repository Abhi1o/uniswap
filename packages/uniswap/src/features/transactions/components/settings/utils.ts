import type { Platform } from 'uniswap/src/features/platforms/types/Platform'
import type { TransactionSettingConfig } from 'uniswap/src/features/transactions/components/settings/types'

/**
 * Filters transaction settings to only include those applicable to the specified platform
 * @param settings Array of transaction setting configs
 * @param platform The platform to filter for
 * @returns Filtered array of settings applicable to the platform
 */
export function filterSettingsByPlatform(
  settings: TransactionSettingConfig[],
  platform: Platform,
): TransactionSettingConfig[] {
  return settings.filter((setting) => setting.applicablePlatforms.includes(platform))
}
