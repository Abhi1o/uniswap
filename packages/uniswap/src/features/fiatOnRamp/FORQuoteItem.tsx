import { useTranslation } from 'react-i18next'
import { Flex, Loader, Text, TouchableArea, UniversalImage, useIsDarkMode } from 'ui/src'
import { TimePast } from 'ui/src/components/icons/TimePast'
import { borderRadii, iconSizes } from 'ui/src/theme'
import { FORServiceProvider } from 'uniswap/src/features/fiatOnRamp/types'
import { getOptionalServiceProviderLogo } from 'uniswap/src/features/fiatOnRamp/utils'

function LogoLoader(): JSX.Element {
  return <Loader.Box borderRadius="$roundedFull" height={iconSizes.icon32} width={iconSizes.icon32} />
}

export function FORQuoteItem({
  serviceProvider,
  onPress,
  isLoading,
  showPaymentMethods = true,
  isRecent = false,
  hidden = false,
}: {
  serviceProvider: FORServiceProvider | undefined
  onPress: () => void
  isLoading?: boolean
  showPaymentMethods?: boolean
  isRecent?: boolean
  hidden?: boolean
}): JSX.Element | null {
  const { t } = useTranslation()
  const isDarkMode = useIsDarkMode()
  const logoUrl = getOptionalServiceProviderLogo(serviceProvider?.logos, isDarkMode)

  if (!serviceProvider) {
    return null
  }

  const paymentMethods =
    serviceProvider.paymentMethods.length > 4
      ? t('fiatOnRamp.quote.type.list', { optionsList: serviceProvider.paymentMethods.slice(0, 3).join(', ') })
      : serviceProvider.paymentMethods.join(', ')

  return (
    <TouchableArea
      disabled={isLoading || hidden}
      disabledStyle={{
        cursor: 'wait',
      }}
      opacity={hidden ? 0 : 1}
      height={hidden ? 0 : 'unset'}
      animation="200ms"
      position={hidden ? 'absolute' : 'relative'}
      onPress={isLoading || hidden ? undefined : onPress}
    >
      <Flex
        backgroundColor="$surface1"
        borderColor="$surface3"
        borderRadius="$rounded20"
        borderWidth="$spacing1"
        hoverStyle={{ backgroundColor: '$surface1Hovered' }}
        p="$spacing16"
        style={{ transition: 'background-color 0.2s ease-in-out' }}
      >
        <Flex row alignItems="center" justifyContent="space-between">
          <Flex row alignItems="center" gap="$spacing12" width="100%">
            <Flex opacity={hidden ? 0 : 1} animation="100ms">
              {logoUrl ? (
                <UniversalImage
                  fallback={<LogoLoader />}
                  size={{
                    height: iconSizes.icon32,
                    width: iconSizes.icon32,
                  }}
                  style={{
                    image: { borderRadius: borderRadii.rounded8 },
                  }}
                  uri={logoUrl}
                />
              ) : (
                <LogoLoader />
              )}
            </Flex>
            <Flex flex={1} gap="$spacing4">
              <Flex row alignItems="center" justifyContent="space-between">
                <Text color="$neutral1" variant="body2">
                  {serviceProvider.name}
                </Text>
                {isRecent && (
                  <Flex row alignItems="center" gap="$spacing8" opacity={showPaymentMethods ? 1 : 0} animation="100ms">
                    <Text color="$neutral2" variant="body4">
                      {t('common.recent')}
                    </Text>
                    <TimePast color="$neutral2" size="$icon.16" />
                  </Flex>
                )}
              </Flex>
              {paymentMethods && (
                <Text
                  color="$neutral2"
                  variant="body4"
                  opacity={showPaymentMethods ? 1 : 0}
                  maxHeight={showPaymentMethods ? 'max-content' : 0}
                  animation="100ms"
                >
                  {paymentMethods}
                </Text>
              )}
            </Flex>
          </Flex>
        </Flex>
      </Flex>
    </TouchableArea>
  )
}
