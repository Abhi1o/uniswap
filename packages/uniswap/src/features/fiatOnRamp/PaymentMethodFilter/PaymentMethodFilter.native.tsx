import { FlatList } from 'react-native'
import {
  PaymentMethodFilterProps,
  useEnabledPaymentMethodFilters,
  useRenderPaymentMethod,
  useTogglePaymentMethod,
} from 'uniswap/src/features/fiatOnRamp/PaymentMethodFilter/utils'
import { FORFilters } from 'uniswap/src/features/fiatOnRamp/types'
import { FiatOffRampEventName, FiatOnRampEventName } from 'uniswap/src/features/telemetry/constants'
import { sendAnalyticsEvent } from 'uniswap/src/features/telemetry/send'

export function PaymentMethodFilter({
  quotes,
  paymentMethod,
  setPaymentMethod,
  isOffRamp,
}: PaymentMethodFilterProps): JSX.Element {
  const enabledPaymentMethodFilters = useEnabledPaymentMethodFilters(quotes)
  const handleTogglePaymentMethod = useTogglePaymentMethod(paymentMethod, setPaymentMethod)
  const handleOnPress = (filter: FORFilters): void => {
    handleTogglePaymentMethod(filter)
    if (isOffRamp) {
      sendAnalyticsEvent(FiatOffRampEventName.FiatOffRampPaymentMethodFilterSelected, { paymentMethodFilter: filter })
    } else {
      sendAnalyticsEvent(FiatOnRampEventName.FiatOnRampPaymentMethodFilterSelected, { paymentMethodFilter: filter })
    }
  }
  const renderPaymentMethod = useRenderPaymentMethod(paymentMethod, handleOnPress)

  return (
    <FlatList
      horizontal
      data={enabledPaymentMethodFilters}
      renderItem={renderPaymentMethod}
      keyExtractor={(item) => item.filter}
      showsHorizontalScrollIndicator={false}
      contentContainerStyle={{ gap: 8, paddingHorizontal: 24 }}
    />
  )
}
