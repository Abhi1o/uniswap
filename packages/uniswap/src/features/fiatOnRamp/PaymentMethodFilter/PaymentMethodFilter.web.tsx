import { Flex, FlexProps } from 'ui/src'
import {
  PaymentMethodFilterProps,
  PaymentMethodItem,
  useEnabledPaymentMethodFilters,
  useTogglePaymentMethod,
} from 'uniswap/src/features/fiatOnRamp/PaymentMethodFilter/utils'
import { FORFilters } from 'uniswap/src/features/fiatOnRamp/types'
import { FiatOffRampEventName, FiatOnRampEventName } from 'uniswap/src/features/telemetry/constants'
import { sendAnalyticsEvent } from 'uniswap/src/features/telemetry/send'

export function PaymentMethodFilter({
  paymentMethod,
  setPaymentMethod,
  isOffRamp,
  quotes,
  ...rest
}: PaymentMethodFilterProps & FlexProps): JSX.Element {
  const enabledPaymentMethodFilters = useEnabledPaymentMethodFilters(quotes)
  const handleTogglePaymentMethod: (method: FORFilters) => void = useTogglePaymentMethod(
    paymentMethod,
    setPaymentMethod,
  )
  const handleOnPress = (filter: FORFilters): void => {
    handleTogglePaymentMethod(filter)
    if (isOffRamp) {
      sendAnalyticsEvent(FiatOffRampEventName.FiatOffRampPaymentMethodFilterSelected, { paymentMethodFilter: filter })
    } else {
      sendAnalyticsEvent(FiatOnRampEventName.FiatOnRampPaymentMethodFilterSelected, { paymentMethodFilter: filter })
    }
  }

  return (
    <Flex row alignItems="center" gap="$gap8" overflow="scroll" scrollbarWidth="none" {...rest}>
      {enabledPaymentMethodFilters.map(({ filter, icon, label }) => {
        const isSelected = paymentMethod === filter

        return (
          <PaymentMethodItem
            key={filter}
            filter={filter}
            icon={icon}
            label={label}
            isSelected={isSelected}
            onPress={() => handleOnPress(filter)}
          />
        )
      })}
    </Flex>
  )
}
