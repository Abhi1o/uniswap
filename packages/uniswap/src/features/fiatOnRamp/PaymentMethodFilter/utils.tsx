import { useCallback, useMemo } from 'react'
import { useTranslation } from 'react-i18next'
import { Flex, FlexProps, GeneratedIcon, Text, TouchableArea } from 'ui/src'
import { AppleLogo } from 'ui/src/components/icons/AppleLogo'
import { Bank } from 'ui/src/components/icons/Bank'
import { Buy } from 'ui/src/components/icons/Buy'
import { GoogleLogo } from 'ui/src/components/icons/GoogleLogo'
import { PaypalLogo } from 'ui/src/components/icons/PaypalLogo'
import { VenmoLogo } from 'ui/src/components/icons/VenmoLogo'
import { iconSizes } from 'ui/src/theme'
import { FORFilters, FORFiltersMap, FORQuote } from 'uniswap/src/features/fiatOnRamp/types'
import { isAndroid, isIOS, isWebAndroid, isWebIOS } from 'utilities/src/platform'

export type PaymentMethodFilterProps = FlexProps & {
  quotes?: Maybe<FORQuote[]>
  paymentMethod?: FORFilters
  setPaymentMethod: (method?: FORFilters) => void
  isOffRamp: boolean
}

export type PaymentMethodConfig = {
  filter: FORFilters
  icon: GeneratedIcon
  label: string
  isSupported: boolean
}

export function usePaymentMethodConfigs(): PaymentMethodConfig[] {
  const { t } = useTranslation()

  return useMemo(
    () => [
      {
        filter: FORFilters.ApplePay,
        icon: AppleLogo,
        label: t('fiatOnRamp.paymentMethods.applePay'),
        isSupported: isIOS || isWebIOS,
      },
      {
        filter: FORFilters.GooglePay,
        icon: GoogleLogo,
        label: t('fiatOnRamp.paymentMethods.googlePay'),
        isSupported: isAndroid || isWebAndroid,
      },
      {
        filter: FORFilters.Bank,
        icon: Bank,
        label: t('fiatOnRamp.paymentMethods.bank'),
        isSupported: true,
      },
      {
        filter: FORFilters.Debit,
        icon: Buy,
        label: t('fiatOnRamp.paymentMethods.debit'),
        isSupported: true,
      },
      {
        filter: FORFilters.PayPal,
        icon: PaypalLogo,
        label: t('fiatOnRamp.paymentMethods.paypal'),
        isSupported: true,
      },
      {
        filter: FORFilters.Venmo,
        icon: VenmoLogo,
        label: t('fiatOnRamp.paymentMethods.venmo'),
        isSupported: true,
      },
    ],
    [t],
  )
}

export function useEnabledPaymentMethodFilters(quotes?: Maybe<FORQuote[]>): PaymentMethodConfig[] {
  const paymentMethodConfigs = usePaymentMethodConfigs()

  return useMemo(() => {
    const useablePaymentMethods = quotes
      ?.flatMap((quote) => quote.serviceProviderDetails.paymentMethods)
      .flatMap((paymentMethod) => {
        const mappedFilter = FORFiltersMap[paymentMethod]
        return mappedFilter
      })

    return paymentMethodConfigs.filter(({ filter, isSupported }) => {
      return useablePaymentMethods?.includes(filter) && isSupported
    })
  }, [quotes, paymentMethodConfigs])
}

export function useTogglePaymentMethod(
  paymentMethod: FORFilters | undefined,
  setPaymentMethod: (method?: FORFilters) => void,
): (method: FORFilters) => void {
  return useCallback(
    (method: FORFilters) => {
      setPaymentMethod(paymentMethod === method ? undefined : method)
    },
    [paymentMethod, setPaymentMethod],
  )
}

type PaymentMethodItemProps = {
  filter: FORFilters
  icon: GeneratedIcon
  label: string
  isSelected: boolean
  onPress: () => void
}

export function PaymentMethodItem({
  filter,
  icon: Icon,
  label,
  isSelected,
  onPress,
}: PaymentMethodItemProps): JSX.Element {
  return (
    <TouchableArea key={filter} scaleTo={0.95} onPress={onPress}>
      <Flex
        row
        alignItems="center"
        gap="$gap4"
        p="$spacing8"
        pl="$spacing4"
        backgroundColor={isSelected ? '$surface3' : '$surface1'}
        borderColor="$surface3"
        borderWidth={1}
        borderRadius="$rounded12"
      >
        <Icon size={iconSizes.icon20} color="$neutral1" />
        <Text variant="body3" color="$neutral1">
          {label}
        </Text>
      </Flex>
    </TouchableArea>
  )
}

export function useRenderPaymentMethod(
  paymentMethod: FORFilters | undefined,
  handleTogglePaymentMethod: (method: FORFilters) => void,
): ({ item }: { item: PaymentMethodConfig }) => JSX.Element {
  return useCallback(
    ({ item: { filter, icon, label } }: { item: PaymentMethodConfig }) => {
      const isSelected = paymentMethod === filter

      return (
        <PaymentMethodItem
          filter={filter}
          icon={icon}
          label={label}
          isSelected={isSelected}
          onPress={() => handleTogglePaymentMethod(filter)}
        />
      )
    },
    [paymentMethod, handleTogglePaymentMethod],
  )
}
