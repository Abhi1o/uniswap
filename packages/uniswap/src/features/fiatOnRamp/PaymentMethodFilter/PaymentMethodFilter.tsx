import { PaymentMethodFilterProps } from 'uniswap/src/features/fiatOnRamp/PaymentMethodFilter/utils'
import { PlatformSplitStubError } from 'utilities/src/errors'

export function PaymentMethodFilter(_props: PaymentMethodFilterProps): JSX.Element {
  throw new PlatformSplitStubError('PaymentMethodFilter')
}
