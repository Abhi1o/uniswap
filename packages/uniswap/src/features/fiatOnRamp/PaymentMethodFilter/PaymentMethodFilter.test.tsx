import { Bank } from 'ui/src/components/icons/Bank'
import { Buy } from 'ui/src/components/icons/Buy'
import { PaymentMethodFilter as MobilePaymentMethodFilter } from 'uniswap/src/features/fiatOnRamp/PaymentMethodFilter/PaymentMethodFilter.native'
import { PaymentMethodFilter as WebPaymentMethodFilter } from 'uniswap/src/features/fiatOnRamp/PaymentMethodFilter/PaymentMethodFilter.web'
import { PaymentMethodItem } from 'uniswap/src/features/fiatOnRamp/PaymentMethodFilter/utils'
import { FORFilters } from 'uniswap/src/features/fiatOnRamp/types'
import { render } from 'uniswap/src/test/test-utils'

jest.mock('utilities/src/platform', () => ({
  isAndroid: jest.fn(),
  isIOS: jest.fn(),
  isWebAndroid: jest.fn(),
  isWebIOS: jest.fn(),
}))

jest.mock('react-i18next', () => ({
  useTranslation: (): { t: (key: string) => string } => ({
    t: (key: string): string => {
      const translations: Record<string, string> = {
        'fiatOnRamp.paymentMethods.applePay': 'Apple Pay',
        'fiatOnRamp.paymentMethods.googlePay': 'Google Pay',
        'fiatOnRamp.paymentMethods.bank': 'Bank',
        'fiatOnRamp.paymentMethods.debit': 'Debit',
        'fiatOnRamp.paymentMethods.paypal': 'PayPal',
        'fiatOnRamp.paymentMethods.venmo': 'Venmo',
      }
      return translations[key] || key
    },
  }),
}))

describe('PaymentMethodFilter', () => {
  const defaultProps = {
    paymentMethod: undefined,
    setPaymentMethod: jest.fn(),
    isOffRamp: false,
  }

  describe('Mobile Implementation', () => {
    beforeEach(() => {
      const { isAndroid, isIOS, isWebAndroid, isWebIOS } = require('utilities/src/platform')
      isAndroid.mockReturnValue(true)
      isIOS.mockReturnValue(false)
      isWebAndroid.mockReturnValue(false)
      isWebIOS.mockReturnValue(false)
    })

    it('renders mobile implementation correctly', () => {
      const tree = render(<MobilePaymentMethodFilter {...defaultProps} />)
      expect(tree).toMatchSnapshot()
    })

    it('renders with selected payment method', () => {
      const propsWithSelection = {
        ...defaultProps,
        paymentMethod: FORFilters.Bank,
      }
      const tree = render(<MobilePaymentMethodFilter {...propsWithSelection} />)
      expect(tree).toMatchSnapshot()
    })

    it('renders with iOS supported payment methods', () => {
      const { isAndroid, isIOS } = require('utilities/src/platform')
      isAndroid.mockReturnValue(false)
      isIOS.mockReturnValue(true)

      const tree = render(<MobilePaymentMethodFilter {...defaultProps} />)
      expect(tree).toMatchSnapshot()
    })
  })

  describe('Web Implementation', () => {
    beforeEach(() => {
      const { isAndroid, isIOS, isWebAndroid, isWebIOS } = require('utilities/src/platform')
      isAndroid.mockReturnValue(false)
      isIOS.mockReturnValue(false)
      isWebAndroid.mockReturnValue(false)
      isWebIOS.mockReturnValue(false)
    })

    it('renders web implementation correctly', () => {
      const tree = render(<WebPaymentMethodFilter {...defaultProps} />)
      expect(tree).toMatchSnapshot()
    })

    it('renders with selected payment method', () => {
      const propsWithSelection = {
        ...defaultProps,
        paymentMethod: FORFilters.PayPal,
      }
      const tree = render(<WebPaymentMethodFilter {...propsWithSelection} />)
      expect(tree).toMatchSnapshot()
    })

    it('renders with web-specific props', () => {
      const propsWithWebProps = {
        ...defaultProps,
        testID: 'web-filter',
        width: 24,
      }
      const tree = render(<WebPaymentMethodFilter {...propsWithWebProps} />)
      expect(tree).toMatchSnapshot()
    })

    it('renders with iOS supported payment methods', () => {
      const { isWebAndroid, isWebIOS } = require('utilities/src/platform')
      isWebAndroid.mockReturnValue(false)
      isWebIOS.mockReturnValue(true)

      const tree = render(<WebPaymentMethodFilter {...defaultProps} />)
      expect(tree).toMatchSnapshot()
    })

    it('renders with Android supported payment methods', () => {
      const { isWebAndroid, isWebIOS } = require('utilities/src/platform')
      isWebAndroid.mockReturnValue(true)
      isWebIOS.mockReturnValue(false)

      const tree = render(<WebPaymentMethodFilter {...defaultProps} />)
      expect(tree).toMatchSnapshot()
    })
  })

  describe('Shared Components', () => {
    it('renders PaymentMethodItem correctly', () => {
      const props = {
        filter: FORFilters.Bank,
        icon: Bank,
        label: 'Bank',
        isSelected: true,
        onPress: jest.fn(),
      }
      const tree = render(<PaymentMethodItem {...props} />)
      expect(tree).toMatchSnapshot()
    })

    it('renders PaymentMethodItem when not selected', () => {
      const props = {
        filter: FORFilters.Debit,
        icon: Buy,
        label: 'Debit',
        isSelected: false,
        onPress: jest.fn(),
      }
      const tree = render(<PaymentMethodItem {...props} />)
      expect(tree).toMatchSnapshot()
    })
  })
})
