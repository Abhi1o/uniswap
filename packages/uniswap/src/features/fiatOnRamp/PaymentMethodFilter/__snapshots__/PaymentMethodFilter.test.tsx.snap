// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentMethodFilter Mobile Implementation renders mobile implementation correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "gap": 8,
      "paddingHorizontal": 24,
    }
  }
  data={[]}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={true}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={0.0001}
  showsHorizontalScrollIndicator={false}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View />
</RCTScrollView>
`;

exports[`PaymentMethodFilter Mobile Implementation renders with iOS supported payment methods 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "gap": 8,
      "paddingHorizontal": 24,
    }
  }
  data={[]}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={true}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={0.0001}
  showsHorizontalScrollIndicator={false}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View />
</RCTScrollView>
`;

exports[`PaymentMethodFilter Mobile Implementation renders with selected payment method 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "gap": 8,
      "paddingHorizontal": 24,
    }
  }
  data={[]}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={true}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={0.0001}
  showsHorizontalScrollIndicator={false}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View />
</RCTScrollView>
`;

exports[`PaymentMethodFilter Shared Components renders PaymentMethodItem correctly 1`] = `
<View
  cancelable={true}
  collapsable={false}
  disabled={false}
  focusVisibleStyle={{}}
  focusable={true}
  forwardedRef={[Function]}
  jestAnimatedStyle={
    {
      "value": {},
    }
  }
  minPressDuration={0}
  onBlur={[Function]}
  onFocus={[Function]}
  onLayout={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  role="button"
  style={
    {
      "backgroundColor": "transparent",
      "borderBottomLeftRadius": 12,
      "borderBottomRightRadius": 12,
      "borderTopLeftRadius": 12,
      "borderTopRightRadius": 12,
      "flexDirection": "column",
      "opacity": 1,
      "transform": [
        {
          "scale": 1,
        },
      ],
    }
  }
>
  <View
    onBlur={[Function]}
    onFocus={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(255,255,255,0.12)",
        "borderBottomColor": "rgba(255,255,255,0.12)",
        "borderBottomLeftRadius": 12,
        "borderBottomRightRadius": 12,
        "borderBottomWidth": 1,
        "borderLeftColor": "rgba(255,255,255,0.12)",
        "borderLeftWidth": 1,
        "borderRightColor": "rgba(255,255,255,0.12)",
        "borderRightWidth": 1,
        "borderStyle": "solid",
        "borderTopColor": "rgba(255,255,255,0.12)",
        "borderTopLeftRadius": 12,
        "borderTopRightRadius": 12,
        "borderTopWidth": 1,
        "flexDirection": "row",
        "gap": 4,
        "paddingBottom": 8,
        "paddingLeft": 4,
        "paddingRight": 8,
        "paddingTop": 8,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={20}
      bbWidth={20}
      fill="none"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      strokeWidth={8}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "color": "#FFFFFF",
            "height": 20,
            "width": 20,
          },
          {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      vbHeight={20}
      vbWidth={20}
    >
      <RNSVGGroup
        color="#FFFFFF"
        fill={null}
        propList={
          [
            "fill",
            "strokeWidth",
          ]
        }
        strokeWidth={8}
      >
        <RNSVGPath
          d="M2.5 6.36549C2.5 6.06293 2.67292 5.78699 2.94519 5.65505L8.96698 2.73705C9.61922 2.42098 10.3803 2.42099 11.0325 2.73705L17.0543 5.65505C17.3266 5.78699 17.4995 6.06293 17.4995 6.36549V7.26185C17.4995 7.27251 17.4993 7.28312 17.4989 7.29368C17.4993 7.30381 17.4995 7.31399 17.4995 7.32423C17.4995 7.72565 17.1741 8.05106 16.7727 8.05106H16.7294L16.7101 8.0513H3.28945L3.27012 8.05106H3.22684C2.82542 8.05106 2.5 7.72565 2.5 7.32423C2.5 7.31399 2.50021 7.30381 2.50063 7.29368C2.50021 7.28312 2.5 7.27251 2.5 7.26185V6.36549Z"
          fill={
            {
              "type": 2,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
        <RNSVGPath
          d="M4.03958 14.5926C3.60358 14.5926 3.25013 14.9461 3.25013 15.3821V16.7106C3.25013 17.1466 3.60358 17.5 4.03958 17.5H15.9602C16.3962 17.5 16.7497 17.1466 16.7497 16.7106V15.3821C16.7497 14.9461 16.3962 14.5926 15.9602 14.5926H4.03958Z"
          fill={
            {
              "type": 2,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
        <RNSVGPath
          d="M6.13177 8.97352C5.69577 8.97352 5.34232 9.32697 5.34232 9.76297V12.9208C5.34232 13.3568 5.69577 13.7102 6.13177 13.7102H6.28966C6.72566 13.7102 7.0791 13.3568 7.0791 12.9208V9.76297C7.0791 9.32697 6.72566 8.97352 6.28966 8.97352H6.13177Z"
          fill={
            {
              "type": 2,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
        <RNSVGPath
          d="M9.13151 9.76297C9.13151 9.32697 9.48496 8.97352 9.92096 8.97352H10.0788C10.5148 8.97352 10.8683 9.32697 10.8683 9.76297V12.9208C10.8683 13.3568 10.5148 13.7102 10.0788 13.7102H9.92096C9.48496 13.7102 9.13151 13.3568 9.13151 12.9208V9.76297Z"
          fill={
            {
              "type": 2,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
        <RNSVGPath
          d="M13.7102 8.97352C13.2742 8.97352 12.9207 9.32697 12.9207 9.76297V12.9208C12.9207 13.3568 13.2742 13.7102 13.7102 13.7102H13.868C14.304 13.7102 14.6575 13.3568 14.6575 12.9208V9.76297C14.6575 9.32697 14.304 8.97352 13.868 8.97352H13.7102Z"
          fill={
            {
              "type": 2,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.4}
      onBlur={[Function]}
      onFocus={[Function]}
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "Basel Grotesk",
          "fontSize": 15,
          "fontWeight": "400",
          "lineHeight": 19.5,
        }
      }
      suppressHighlighting={true}
    >
      Bank
    </Text>
  </View>
</View>
`;

exports[`PaymentMethodFilter Shared Components renders PaymentMethodItem when not selected 1`] = `
<View
  cancelable={true}
  collapsable={false}
  disabled={false}
  focusVisibleStyle={{}}
  focusable={true}
  forwardedRef={[Function]}
  jestAnimatedStyle={
    {
      "value": {},
    }
  }
  minPressDuration={0}
  onBlur={[Function]}
  onFocus={[Function]}
  onLayout={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  role="button"
  style={
    {
      "backgroundColor": "transparent",
      "borderBottomLeftRadius": 12,
      "borderBottomRightRadius": 12,
      "borderTopLeftRadius": 12,
      "borderTopRightRadius": 12,
      "flexDirection": "column",
      "opacity": 1,
      "transform": [
        {
          "scale": 1,
        },
      ],
    }
  }
>
  <View
    onBlur={[Function]}
    onFocus={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#131313",
        "borderBottomColor": "rgba(255,255,255,0.12)",
        "borderBottomLeftRadius": 12,
        "borderBottomRightRadius": 12,
        "borderBottomWidth": 1,
        "borderLeftColor": "rgba(255,255,255,0.12)",
        "borderLeftWidth": 1,
        "borderRightColor": "rgba(255,255,255,0.12)",
        "borderRightWidth": 1,
        "borderStyle": "solid",
        "borderTopColor": "rgba(255,255,255,0.12)",
        "borderTopLeftRadius": 12,
        "borderTopRightRadius": 12,
        "borderTopWidth": 1,
        "flexDirection": "row",
        "gap": 4,
        "paddingBottom": 8,
        "paddingLeft": 4,
        "paddingRight": 8,
        "paddingTop": 8,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={20}
      bbWidth={20}
      fill="none"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      strokeWidth={8}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "color": "#FFFFFF",
            "height": 20,
            "width": 20,
          },
          {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
    >
      <RNSVGGroup
        color="#FFFFFF"
        fill={null}
        propList={
          [
            "fill",
            "strokeWidth",
          ]
        }
        strokeWidth={8}
      >
        <RNSVGPath
          d="M21 8V9.25H3V8C3 6 4 5 6 5H18C20 5 21 6 21 8ZM21 10.75V16C21 18 20 19 18 19H6C4 19 3 18 3 16V10.75H21ZM10.75 15C10.75 14.586 10.414 14.25 10 14.25H7C6.586 14.25 6.25 14.586 6.25 15C6.25 15.414 6.586 15.75 7 15.75H10C10.414 15.75 10.75 15.414 10.75 15Z"
          fill={
            {
              "type": 2,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.4}
      onBlur={[Function]}
      onFocus={[Function]}
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "Basel Grotesk",
          "fontSize": 15,
          "fontWeight": "400",
          "lineHeight": 19.5,
        }
      }
      suppressHighlighting={true}
    >
      Debit
    </Text>
  </View>
</View>
`;

exports[`PaymentMethodFilter Web Implementation renders web implementation correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "gap": 8,
      "overflow": "scroll",
    }
  }
/>
`;

exports[`PaymentMethodFilter Web Implementation renders with Android supported payment methods 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "gap": 8,
      "overflow": "scroll",
    }
  }
/>
`;

exports[`PaymentMethodFilter Web Implementation renders with iOS supported payment methods 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "gap": 8,
      "overflow": "scroll",
    }
  }
/>
`;

exports[`PaymentMethodFilter Web Implementation renders with selected payment method 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "gap": 8,
      "overflow": "scroll",
    }
  }
/>
`;

exports[`PaymentMethodFilter Web Implementation renders with web-specific props 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "gap": 8,
      "overflow": "scroll",
      "width": 24,
    }
  }
  testID="web-filter"
/>
`;
